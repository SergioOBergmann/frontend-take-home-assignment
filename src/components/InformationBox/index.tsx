import * as React from 'react';
import {
  InformationBoxDiv,
  PrimaryMessage,
  Label,
  Value,
  SecondaryMessage,
  Text
} from './styles';

type InformationBoxProps = {
  desktopPrimaryMessageLabel: string;
  mobilePrimaryMessageLabel: string;
  primaryMessageValue: string;
  secondaryMessage: JSX.Element;
};

const InformationBox = ({
  desktopPrimaryMessageLabel,
  mobilePrimaryMessageLabel,
  primaryMessageValue,
  secondaryMessage
}: InformationBoxProps): JSX.Element => {
  return (
    <>
      <InformationBoxDiv>
        <PrimaryMessage>
          <Label className="desktop">{desktopPrimaryMessageLabel}</Label>
          <Label className="mobile">{mobilePrimaryMessageLabel}</Label>
          <Value>{primaryMessageValue}</Value>
        </PrimaryMessage>
        <SecondaryMessage>
          <Text>{secondaryMessage}</Text>
        </SecondaryMessage>
      </InformationBoxDiv>
    </>
  );
};

export default InformationBox;
