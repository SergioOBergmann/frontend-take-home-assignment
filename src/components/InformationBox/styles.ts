import styled from 'styled-components';

import { MOBILE_BREAKPOINT } from '../../styles/global';

export const InformationBoxDiv = styled.div`
  flex-direction: column;
  height: 168px;
  margin-top: 31px;

  background: #ffffff;
  border: 1px solid #e1e8ed;
  box-shadow: 0px 1px 4px rgba(150, 164, 176, 0.1);
  border-radius: 4px;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    height: 134px;
    margin-top: 30px;
  }
`;

export const PrimaryMessage = styled.div`
  flex-wrap: wrap;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
  height: 102px;
  padding: 0 32px;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    height: 64px;
    padding: 0 24px;
  }
`;

export const Label = styled.span`
  font-weight: 500;

  &.mobile {
    display: none;
  }

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    font-size: 16px;
    &.desktop {
      display: none;
    }
    &.mobile {
      display: block;
    }
  }
`;

export const Value = styled.span`
  flex-grow: 0;
  max-width: 407px;

  font-weight: 500;
  font-size: 40px;
  line-height: 40px;

  overflow: hidden;
  text-overflow: ellipsis;
  text-align: center;
  letter-spacing: -0.266667px;

  color: #0079FF;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    max-width: 264px
    font-size: 26px;
    line-height: 32px;
    letter-spacing: -0.216667px;
  }
`;

export const SecondaryMessage = styled.div`
  flex-direction: row;
  align-items: center;
  height: 66px;
  padding: 0 32px;

  background: #f4f8fa;
  border-radius: 0 0 4px 4px;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    padding: 0 24px;
    height: 70px;
  }
`;

export const Text = styled.label`
  flex: 1;

  font-size: 12px;
  line-height: 16px;
`;
