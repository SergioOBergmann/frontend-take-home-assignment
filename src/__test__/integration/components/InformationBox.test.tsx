import * as React from 'react';

import { render, RenderResult } from '@testing-library/react';

import InformationBox from '../../../components/InformationBox';

let documentBody: RenderResult;

describe('<InformationBox />', () => {
  beforeEach(() => {
    documentBody = render(
      <InformationBox
        desktopPrimaryMessageLabel="Monthly amount"
        mobilePrimaryMessageLabel="Monthly"
        primaryMessageValue="25,000"
        secondaryMessage={
          <>
            {"You're planning "}
            <span>
              <strong>1 monthly deposit</strong>
            </span>
            {' to reach your '}
            <strong>$25,000</strong>
            {' goal by '}
            <strong>November 2020.</strong>
          </>
        }
      />
    );
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
