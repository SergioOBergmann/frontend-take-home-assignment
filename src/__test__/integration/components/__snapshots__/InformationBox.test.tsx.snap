// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InformationBox /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-gzVnrw ejlxmw"
    >
      <div
        class="sc-htoDjs kUNGsP"
      >
        <span
          class="sc-dnqmqq fbLvhl desktop"
        >
          Monthly amount
        </span>
        <span
          class="sc-dnqmqq fbLvhl mobile"
        >
          Monthly
        </span>
        <span
          class="sc-iwsKbI dTSlFv"
        >
          25,000
        </span>
      </div>
      <div
        class="sc-gZMcBi gjjmRo"
      >
        <label
          class="sc-gqjmRU eskmOx"
        >
          You're planning 
          <span>
            <strong>
              1 monthly deposit
            </strong>
          </span>
           to reach your 
          <strong>
            $25,000
          </strong>
           goal by 
          <strong>
            November 2020.
          </strong>
        </label>
      </div>
    </div>
  </div>
</body>
`;
